src/routes/(pages)/blog/posts/[slug]/+page.ts

```diff
@@ -9,13 +12,27 @@ export const load = async ({ url, params }) => {
 
   try {
     const post = await import(`../../../../../blog-posts/${params.slug}.md`);
+    const metadata = metadataSchema.parse(post.metadata);
+
+    let author: z.infer<typeof authorSchema> | undefined;
+    if (post.metadata.author) {
+      const authorDesc = await import(`../../../../../blog-posts/authors/${metadata.author}.json`);
+      assert(
+        authorDesc,
+        `Unable to locate blog author with ID ${metadata.author}. Make sure the ID is present in /src/blog-posts/authors/`,
+      );
+
+      author = authorSchema.parse(authorDesc);
+    }
```

Comment:
Can `getBlogPosts` be reused here?

Reply:
I tried to but for some reason... no! When I externalize the dynamic import outside the load function, there's a server-side crash from somewhere within `mdsvex` with error `process is not defined`. I don't really understand why but don't think it's worth it to dig deeper atm. It must be something about how code within a pre-rendered load function is not ran server-side again, which might not apply to externalized modules? I'm not sure...